app/auth/logout/helpers.ts

```diff
@@ -0,0 +1,80 @@
+import { IronSession } from "iron-session"
+import { NextResponse } from "next/server"
+import * as client from "openid-client"
+
+import goConfig from "@/lib/config/goConfig"
+import { getUniloginClientConfig } from "@/lib/session/oauth/uniloginClient"
+import { TSessionData, deleteUniloginIdToken, getUniloginIdToken } from "@/lib/session/session"
+
+import loadAdgangsplatformenLogoutUrl from "./loadAdgangsplatformenLogoutUrl"
+
+export const destroySessionAndRedirectToFrontPage = async (session: IronSession<TSessionData>) => {
+  const appUrl = new URL(String(goConfig("app.url")))
+
+  const id_token = await getUniloginIdToken()
+  if (id_token) {
+    await deleteUniloginIdToken()
+  }
+
+  session.destroy()
+  return NextResponse.redirect(`${appUrl.toString()}?reload-session=true`)
+}
+
+export const redirectToFrontPage = async () => {
+  const appUrl = new URL(String(goConfig("app.url")))
+
+  return NextResponse.redirect(appUrl)
+}
+
+export const handleUniloginLogout = async (session: IronSession<TSessionData>) => {
+  const config = await getUniloginClientConfig()
+  const id_token = await getUniloginIdToken()
+
+  // If we don't have a client config, we can't do anything.
```

Comment:
I really like these inline comments explaining this long procedure. Well done.

Reply:
thx :)